controller/src/controller.rs

```diff
@@ -64,8 +64,8 @@ fn check_middleware(
 			}
 			if let Some(s) = slate {
 				if s.version_info.version < CURRENT_SLATE_VERSION
-					|| (bhv == 2 && s.version_info.block_header_version != 2)
-					|| (bhv > 2 && s.version_info.block_header_version < GRIN_BLOCK_HEADER_VERSION)
+					|| (bhv == 3 && s.version_info.block_header_version != 3)
```

Comment:
Since txns from block header 2 are compatible with those from block header version 3, you can just remove line 67. We only needed that logic because of the bulletproof serialization changes.

Reply:
:+1: